lib/commands/join.ts

```diff
@@ -72,8 +76,9 @@ export default class JoinCmd extends Command {
 
 	public static flags: flags.Input<FlagsDef> = {
 		application: {
-			description: 'the name of the application the device should join',
 			...cf.application,
+			description:
+				'the name or org/name slug of the application the device should join',
```

Comment:
Right. Actually join does accept an ID - this custom message was left because it mentioned join, but perhaps the standard message should be used.   Changing.